.github/workflows/pytest.yml

```diff
@@ -258,8 +258,7 @@ jobs:
           python --version
           pip --version
           python -m pip install -U pip
-          pip install -r requirements.txt
-          pip install -r requirements_test.txt
+          pip install '.[test]'
```

Comment:
This pulls in all the distributed things as well, if I rmemeber correctly. Is that intentional?

Reply:
Yes, so we can use DDP

Reply:
But why? That's just going to baloon the test runtime to run combinatorial tests with DDP. We already run too many distributed tests (which is why we see so many timeouts).

Reply:
I would like to use DDP because the combinatorial tests is one of the few tests that run end to end model training with different parameters and architectures and that can help us catch issues similar to these.

Reply:
True, but the cost is probably going to be prohibitve it be honest. We're talking orders of magnitude slower because of all the overhead Ray incurs. We can try it, but let's change it back if it ends up being too slow.

Reply:
This run took 25min and before introducing this it takes 17min.